node/sc/bridge_test.go

```diff
@@ -665,3 +665,313 @@ func TestExtendedBridgeAndCallbackERC721(t *testing.T) {
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+
+	// fail locking token if it is already unlocked.
+	tx, err = b.UnlockToken(operator, erc20Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+
+	tx, err = b.UnlockToken(operator, erc721Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+
+	tx, err = b.UnlockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+}
+
+// TestBridgeContract_CheckValueTransferAfterUnLock checks the following:
+// - the token can be unlock to allow value transfer requests.
+func TestBridgeContract_CheckValueTransferAfterUnLock(t *testing.T) {
+	backend, operator, tester, b, erc20, erc721, erc20Addr, erc721Addr := generateBridgeTokenTestEnv(t)
+
+	// lock token
+	tx, err := b.LockToken(operator, erc20Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.LockToken(operator, erc721Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.LockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	// unlock token
+	tx, err = b.UnlockToken(operator, erc20Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.UnlockToken(operator, erc721Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.UnlockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	// check value after unlocking
+	isLocked, err := b.LockedTokens(nil, erc20Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+
+	isLocked, err = b.LockedTokens(nil, erc721Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+
+	isLocked, err = b.IsLockedKLAY(nil)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+
+	// check to allow value transfer
+	tx, err = erc20.RequestValueTransfer(tester, big.NewInt(1), operator.From, big.NewInt(0), nil)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.Nil(t, bind.CheckWaitMined(backend, tx))
+
+	tx, err = erc721.RequestValueTransfer(tester, big.NewInt(1), operator.From, nil)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.Nil(t, bind.CheckWaitMined(backend, tx))
+
+	tester.Value = big.NewInt(1)
+	tx, err = b.RequestKLAYTransfer(tester, tester.From, big.NewInt(0), nil)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.NotNil(t, bind.CheckWaitMined(backend, tx))
```

Comment:
Why it doesn't return an error?

Reply:
Receipt status can be several value. But now there are not all errors corresponded with all receipt status. So it just return success or not like below.

```
// CheckWaitMined returns an error if the transaction is not found or not successful status.
func CheckWaitMined(b DeployBackend, tx *types.Transaction) error {
	timeoutContext, cancelTimeout := context.WithTimeout(context.Background(), 3*time.Second)
	defer cancelTimeout()

	receipt, err := WaitMined(timeoutContext, b, tx)
	if err != nil {
		return err
	}

	if receipt == nil {
		return errors.New("receipt not found")
	}

	if receipt.Status != types.ReceiptStatusSuccessful {
		return errors.New("not successful receipt")
	}
	return nil
}
```

Reply:
I means.... Why it is not `assert.Nil(t, bind.CheckWaitMined(backend, tx))`?
Since the KLAY transfer is unlocked now, the backend should process the transfer succesfully.

Reply:
It's my mistake. :) Thank you very much. I fixed. HAHAHA